### PR TITLE
Location Summary end-point

### PR DESCRIPTION
**Pull request by maxnazaruk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 1]
```diff
@@ -1,59 +1,12 @@
...
 import greencity.client.UserRemoteClient;
 import greencity.constant.ErrorMessage;
 import greencity.constant.OrderHistory;
-import greencity.dto.AllActiveLocationsDto;
-import greencity.dto.CreateAddressRequestDto;
-import greencity.dto.LocationsDtos;
-import greencity.dto.OrderCourierPopUpDto;
-import greencity.dto.RegionDto;
-import greencity.dto.TariffsForLocationDto;
+import greencity.dto.*;
```
Comment (OlegVat) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Please do not use the wildcard imports

File: service-api/src/main/java/greencity/service/ubs/UBSClientService.java [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import greencity.dto.customer.UbsCustomersDto;
 import greencity.dto.customer.UbsCustomersDtoUpdate;
 import greencity.dto.employee.UserEmployeeAuthorityDto;
+import greencity.dto.location.LocationInfoDto;
```
Comment (OlegVat) [inline service-api/src/main/java/greencity/service/ubs/UBSClientService.java]: Please delete the redundant import to get rid of the code smell

File: core/src/main/java/greencity/controller/UserProfileController.java [hunk 3]
```diff
@@ -84,4 +84,21 @@ public ResponseEntity<HttpStatus> deactivateUser(
...
+     * @return {@link ResponseEntity} list of locations.
+     * @author Max Nazaruk.
+     */
+    @ApiOperation(value = "Get info about cities in each region")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = LocationSummaryDto.class),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @GetMapping("/getLocationSummary")
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/UserProfileController.java]: please read: https://restfulapi.net

File: service-api/src/main/java/greencity/dto/location/LocationSummaryDto.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+@Builder
+public class LocationSummaryDto {
+    private Long regionId;
+    private String nameEn;
+    private String nameUa;
+    private List<LocationToCityDto> citiesUa;
+    private List<LocationToCityDto> citiesEn;
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/location/LocationSummaryDto.java:18]: Why you create new DTO when similar is already exist ? If you will need to add field to some DTO, just extend from the existing DTO that contains similar fields and add needed field.

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 5]
```diff
@@ -63,51 +21,20 @@
...
-import greencity.dto.order.OrderAddressDtoRequest;
-import greencity.dto.order.OrderCancellationReasonDto;
-import greencity.dto.order.OrderClientDto;
-import greencity.dto.order.OrderFondyClientDto;
-import greencity.dto.order.OrderPaymentDetailDto;
-import greencity.dto.order.OrderResponseDto;
-import greencity.dto.order.OrderStatusPageDto;
-import greencity.dto.order.OrderWithAddressesResponseDto;
-import greencity.dto.order.OrdersDataForUserDto;
+import greencity.dto.order.*;
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Please configure your intellij to use single class imports.

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 6]
```diff
@@ -63,51 +21,20 @@
...
-import greencity.dto.user.PersonalDataDto;
-import greencity.dto.user.PointsForUbsUserDto;
-import greencity.dto.user.UserInfoDto;
-import greencity.dto.user.UserPointDto;
-import greencity.dto.user.UserPointsAndAllBagsDto;
-import greencity.dto.user.UserProfileDto;
-import greencity.dto.user.UserProfileUpdateDto;
-import greencity.enums.*;
+import greencity.dto.payment.*;
+import greencity.dto.user.*;
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Please configure your intellij to use single class imports.

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 7]
```diff
@@ -1943,4 +1870,11 @@ public Set<String> getAllAuthorities(String email) {
     public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String email) {
         userRemoteClient.updateEmployeesAuthorities(dto, email);
     }
+
+    @Override
+    public List<LocationSummaryDto> getSummaryLocation() {
+        return regionRepository.findAll().stream()
+            .map(i -> modelMapper.map(i, LocationSummaryDto.class))
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: please choose the better name for the local variables

**Review by VadymMakitra — COMMENTED** [review]
Please resolve comments

File: service-api/src/main/java/greencity/dto/location/LocationSummaryDto.java (hunk 4 above)
Comment (maxnazaruk) [reply service-api/src/main/java/greencity/dto/location/LocationSummaryDto.java:18]: There is only one fittable dto present -> LocationDto which contains a single regionId field, no other dtos extend the LoationsDto for regionId. Did in the same way.

**Review by OlegVat — APPROVED** [review]

File: core/src/main/java/greencity/controller/UserProfileController.java (hunk 3 above)
Comment (VadymMakitra) [reply core/src/main/java/greencity/controller/UserProfileController.java]: Still not correct

Comment (maxnazaruk) [reply core/src/main/java/greencity/controller/UserProfileController.java]: Could you please be more specific? What exactly is the problem of this area?

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java (hunk 7 above)
Comment (maxnazaruk) [reply service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: getSummaryLocation -> getLocationSummary

**Review by VadymMakitra — APPROVED** [review]

**Review by innayashna — APPROVED** [review]

**Pull request merged**